evals/cybench/challenges/avatar/compose.yaml

```diff
@@ -0,0 +1,9 @@
+services:
+  default:
+    image: agent-environment:1.0.0
+    init: true
+    environment:
+      VICTIM_HOSTNAME: victim
+  victim:
+    image: avatar-victim:1.0.0
```

Comment:
We ended up needing healthchecks on some of the GDM images to make sure they were "up" before the eval started. We may or may not need them here, just wanted you to know that GDM required it.

Reply:
Yep - this prevents the `docker compose up --wait` command from returning until all services are considered healthy. This means that we don't let the agent get started until everything is up and running.

Here's one example. Note that `curl` would have been cleaner than `wget` if it were available in the container.

https://github.com/UKGovernmentBEIS/inspect_ai/blob/main/evals/gdm_capabilities/in_house_ctf/challenges/cmd_injection/compose.yaml

K8s will use `readinessProbe` for the equivalent functionality.

Reply:
I've added a `healthcheck` and explicitly `expose`d the port to avatar. @sam-deverett-dsit / @pgiav would you be able to do the same for the other challenges? See my comment in this thread above for some more examples.

Reply:
If we add the healthcheck, shouldn't the agent service then also have an entry like:

```yaml
  default:
    image: agent-sandbox:1.0.0
    init: true
    depends_on:
      victim:
        condition: service_healthly
```

Or does that check happen silently elsewhere?

Reply:
Good question. It doesn't happen silently, but as the default command is something trivial like `sleep infinity` or `tail /dev/null`, there is nothing that we have to wait for. Typically, the healthcheck is for making sure than something slow starting (e.g. a web server, a database etc.). So I'm happy without it for `default`. But lmk if you disagree or want to discuss!